tests/Unit/Domain.LinnApps.Tests/PurchaseOrderDeliveryServiceTests/WhenUpdatingDetailsRemovingFromDeliveryTotal.cs

```diff
@@ -0,0 +1,138 @@
+                                                     BaseOurUnitPrice = 11,
+                                                     DateRequested = 25.March(2022),
+                                                     AvailableAtSupplier = "N"
+                                                 }
+                                         };
+
+
+            // Order will be updated to delivery total. Simulate removing deliveries.  
+            this.line = new PurchaseOrderDetail
+            {
+                Line = 1,
+                OrderUnitPriceCurrency = 666,
+                OrderConversionFactor = 1.5m,
+                BaseOurUnitPrice = 111,
+                PurchaseDeliveries = new List<PurchaseOrderDelivery>(),
+                OurQty = 100,
+                OrderQty = 100
+            };
+            this.order = new PurchaseOrder
+            {
+                OrderNumber = 123456,
+                Details = new List<PurchaseOrderDetail> { this.line },
+                SupplierId = 9876543,
+                DocumentTypeName = "PO"
+            };
+            this.PurchaseOrderRepository.FindById(this.order.OrderNumber).Returns(this.order);
+
+            this.PurchaseOrdersPack.GetVatAmountSupplier(
+                this.line.BaseOurUnitPrice.GetValueOrDefault() * this.updatedDeliveries.First().OurDeliveryQty.GetValueOrDefault(),
+                this.order.SupplierId).Returns(this.baseVatAmount);
+
+            this.PurchaseOrdersPack.GetVatAmountSupplier(
+                this.line.OrderUnitPriceCurrency.GetValueOrDefault()
+                * this.updatedDeliveries.First().OurDeliveryQty.GetValueOrDefault(),
+                this.order.SupplierId).Returns(this.vatAmount);
+
+            this.AuthService.HasPermissionFor(
+                    AuthorisedAction.PurchaseOrderUpdate, Arg.Any<IEnumerable<string>>())
+                .Returns(true);
+
+            this.MiniOrderRepository.FindById(this.order.OrderNumber).Returns(new MiniOrder { });
+            this.result = this.Sut.UpdateDeliveriesForOrderLine(
+                this.order.OrderNumber,
+                this.line.Line,
+                this.updatedDeliveries,
+                new List<string>());
+        }
+
+        [Test]
+        public void ShouldReturnResult()
+        {
+            //Delivery fields
+            this.result.Count().Should().Be(1);
+            var updateData = this.updatedDeliveries.First();
+            var updated = this.result.First();
+            updated.DeliverySeq.Should().Be(1);
+            updated.OurDeliveryQty.Should().Be(updateData.OurDeliveryQty);
+            updated.OrderDeliveryQty.Should().Be(updateData.OurDeliveryQty / this.line.OrderConversionFactor);
+            updated.OurUnitPriceCurrency.Should().Be(this.line.OurUnitPriceCurrency);
+            updated.DateRequested.Should().Be(updateData.DateRequested);
+            updated.DateAdvised.Should().Be(updateData.DateAdvised);
```

Comment:
again you're needlessly testing a whole bunch of fields here that are irrelevant to the scenario of adding/removing deliveries. I'd just test the ones that matter (order line qty and related total values)